### PR TITLE
Move TitleViewSetOnShellWorksAfterNavigation to Appium

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change

TitleViewSetOnShellWorksAfterNavigation is being a bit flakey on device tests so I'm moving this to Appium. Typically device tests flakiness happens because of the complexity of wiring up an entire fake application using our device tests. We mainly were doing this because we didn't really have appium setup in the beginning, but now that we have Appium it's more reliable and accurate to just host the test there.

### CODE REVIEW

**Review by rmarinho — CHANGES_REQUESTED** [review]
The test is failing

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue12888.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	{
+		public Issue12888(TestDevice testDevice) : base(testDevice)
+		{
+		}
+
+		public override string Issue => "Fix timing on iOS Toolbar";
+
+		[Category(UITestCategories.Shell)]
+		[Test(Description = "TitleView Set On Shell Works After Navigation")]
+		public void TitleViewSetOnShellWorksAfterNavigation()
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue12888.cs:17]: Fails on Catalyst and iOS:

```
System.TimeoutException : Timed out waiting for element...

at UITest.Appium.HelperExtensions.Wait(Func`1 query, Func`2 satisfactory, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency) in /_/src/TestUtils/src/UITest.Appium/HelperExtensions.cs:line 1050
   at UITest.Appium.HelperExtensions.WaitForAtLeastOne(Func`1 query, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency) in /_/src/TestUtils/src/UITest.Appium/HelperExtensions.cs:line 1065
   at UITest.Appium.HelperExtensions.WaitForElement(IApp app, String marked, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency, Nullable`1 postTimeout) in /_/src/TestUtils/src/UITest.Appium/HelperExtensions.cs:line 371
   at Microsoft.Maui.TestCases.Tests.Issues.Issue12888.TitleViewSetOnShellWorksAfterNavigation() in /_/src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue12888.cs:line 19
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
```

**Pull request merged**